packages/components/Tabs/src/TabsItem.tsx

```diff
@@ -0,0 +1,158 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { View } from 'react-native';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { Text } from '@fluentui-react-native/text';
+import { Icon } from '@fluentui-react-native/icon';
+import { settings, tabsItemSelectActionLabel } from './TabsItem.settings';
+import { backgroundColorTokens, borderTokens, textTokens, foregroundColorTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
+import { filterViewProps } from '@fluentui-react-native/adapters';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { TabsContext } from './Tabs';
+import { tabsItemName, TabsItemType, TabsItemProps, TabsItemSlotProps, TabsItemRenderData, TabsItemState } from './TabsItem.types';
+import {
+  useAsPressable,
+  useKeyCallback,
+  useViewCommandFocus,
+  createIconProps,
+  useOnPressWithFocus,
+} from '@fluentui-react-native/interactive-hooks';
+
+export const TabsItem = compose<TabsItemType>({
+  displayName: tabsItemName,
+
+  usePrepareProps: (userProps: TabsItemProps, useStyling: IUseComposeStyling<TabsItemType>) => {
+    const {
+      icon,
+      headerText = '',
+      onAccessibilityTap = userProps.onClick,
+      accessibilityLabel = userProps.headerText,
+      componentRef = React.useRef(null),
+      testID,
+      onClick,
+      itemKey,
+      itemCount,
+      ...rest
+    } = userProps;
+
+    // Grabs the context information from Tabs (currently selected TabsItem and client's onTabsClick callback)
+    const info = React.useContext(TabsContext);
+
+    /* We don't want to call the user's onTabsClick multiple times on the same selection. */
```

Comment:
This is more of an opinion but I'd personally err on being "dumb," for lack of a better word, and send all events as we get them. We should let the user decide how to handle duplicate clicks and not filter events out preemptively, in case they want that information (i.e. multiple clicks may indicate a perf issue).

That said I'm not sure how "opinionated" FURN usually is so I'd be curious to hear other people's thoughts.

Reply:
It looks like this specific piece of code was taken from RadioButton. If I remember correctly (this was a long time ago), the user callback was being called multiple times on **one** click (don't quite remember why, I think it had to deal with changing selection on both onFocus and onPress)

It looks like something similar is going on here, where there's two slightly different things going on here between onPress and onFocus.

Reply:
I see. I think it might be worth updating the comment to be more specific about the issue that's being avoided. at least to me it currently reads more like a behavior decision than a fix for a bug.

Reply:
@samuelfreiberg do you have a better way of wording this?

Reply:
@dake3601 I would probably want something like:

"There's a bug where the user callback is being called multiple times on one click. We check that there is an actual change in selection before forwarding the message to the user callback so that we don't process more than one callback for each click"